Admin/Admin.php

```diff
@@ -923,11 +923,37 @@ public function buildDatagrid()
      * Returns the name of the parent related field, so the field can be use to set the default
      * value (ie the parent object) or to filter the object.
      *
-     * @return string the name of the parent related field
+     * @return null|string
+     *
+     * @throws \InvalidArgumentException
      */
     public function getParentAssociationMapping()
     {
-        return $this->parentAssociationMapping;
+        if (is_array($this->parentAssociationMapping) && $this->getParent()) {
+            $parent = $this->getParent()->getCode();
+            if (array_key_exists($parent, $this->parentAssociationMapping)) {
+                return $this->parentAssociationMapping[$parent];
+            } else {
+                throw new \InvalidArgumentException(sprintf(
+                    "There's no association between %s and %s.",
+                    $this->getCode(),
+                    $this->getParent()->getCode()
+                ));
+            }
+        } else {
+            return $this->parentAssociationMapping;
+        }
+    }
+
+    /**
+     * @param string $code
+     * @param string $value
+     *
+     * @return $this
+     */
+    public function addParentAssociationMapping($code, $value)
+    {
+        $this->parentAssociationMapping[$code] = $value;
```

Comment:
I am not available to fix this until Monday, can you wait? I will do first time in the morning as soon as I go to my office.

Enviado desde mi iPhone

> El 9 abr 2016, a las 14:51, Grégoire Paris <notifications@github.com> escribió:
> 
> In Admin/Admin.php:
> 
> > +                ));
> > +            }
> > +        } else {
> > +            return $this->parentAssociationMapping;
> > +        }
> > +    }
> > +
> > +    /**
> > +     * @param string $code
> > +     * @param string $value
> > +     *
> > +     * @return $this
> > +     */
> > +    public function addParentAssociationMapping($code, $value)
> > +    {
> > +        $this->parentAssociationMapping[$code] = $value;
> @robhunt3r, can you fix that quickly ? We are trying to merge all these PRs before doing some big changes on the project.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>